src/plugins/chart_expressions/expression_xy/public/components/legend_action.tsx

```diff
@@ -36,36 +37,31 @@ export const getLegendAction = (
     }
 
     const layer = dataLayers[layerIndex];
-    if (!layer || !layer.splitAccessor) {
+    if (!layer || !layer.splitAccessors || !layer.splitAccessors.length) {
       return null;
     }
 
-    const splitLabel = series.seriesKeys[0] as string;
-
     const { table } = layer;
-    const accessor = getAccessorByDimension(layer.splitAccessor, table.columns);
-    const formatter = formatFactory(
-      accessor ? getFormat(table.columns, layer.splitAccessor) : undefined
-    );
 
-    const rowIndex = table.rows.findIndex((row) => {
-      if (formattedDatatables[layer.layerId]?.formattedColumns[accessor]) {
-        // stringify the value to compare with the chart value
-        return formatter.convert(row[accessor]) === splitLabel;
+    const data: FilterEvent['data']['data'] = [];
+
+    series.splitAccessors.forEach((value, key) => {
```

Comment:
Here splitAccessors is Map. We don't have reduce for Map in js

Reply:
We are doing the mutation of the object from the upper scope. According to the Kibana styleguide, it is necessary to avoid such situations. https://github.com/elastic/kibana/blob/main/STYLEGUIDE.mdx#avoid-mutability-and-state.
```js
// good
function addBar(foos, foo) {
  const newFoo = { ...foo, name: 'bar' };
  return [...foos, newFoo];
}

// bad
function addBar(foos, foo) {
  foo.name = 'bar';
  foos.push(foo);
}
```

So, to avoid such a situation, it would be great to get rid of all the mutations, which are present in this part of the code.

Possible solution:
![Screenshot 2022-07-04 at 17 32 13](https://user-images.githubusercontent.com/22456368/177175345-e18fe312-bca6-4511-9c33-eeb9f76799a9.png)

Reply:
done